packages/terra-date-picker/src/DateRange.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react';
+import DatePicker from './DatePicker';
+
+const propTypes = {
+  /**
+   * A moment object to use as the default end date for a date range.
+   */
+  endDate: PropTypes.oneOfType([PropTypes.object]),
+  /**
+   * A moment object to use as the default start date for a date range.
+   */
+  startDate: PropTypes.oneOfType([PropTypes.object]),
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   */
+  onChange: PropTypes.func,
+};
+
+class DateRange extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: props.startDate,
+      endDate: props.endDate,
+    };
+    this.handleChangeStart = this.handleChangeStart.bind(this);
+    this.handleChangeEnd = this.handleChangeEnd.bind(this);
+  }
+
+  handleChange({ startDate, endDate }) {
+    let startDateForRange = startDate || this.state.startDate;
+    let endDateForRange = endDate || this.state.endDate;
+
+    if (startDateForRange.isAfter(endDateForRange)) {
+      const tempDate = startDateForRange;
+      startDateForRange = endDateForRange;
+      endDateForRange = tempDate;
+    }
+
+    this.setState({ startDate: startDateForRange, endDate: endDateForRange });
+
+    if (this.props.onChange) {
+      this.props.onChange(startDateForRange, endDateForRange);
+    }
+  }
+
+  handleChangeStart(startDate) {
+    this.handleChange({ startDate });
+  }
+
+  handleChangeEnd(endDate) {
+    this.handleChange({ endDate });
+  }
+
+  render() {
+    return (<div className="terra-DatePicker-range">
+      <DatePicker
+        {...this.props}
```

Comment:
Looks like there are some packages that put customProps first and some put it last. I'll bring this up in the war room so they can be consistent. I agree it should be last and updated it. https://github.com/cerner/terra-core/commit/c599197256ce3c511ea206ca5e87c4015f4f6fee

First:
https://github.com/cerner/terra-core/blob/master/packages/terra-progress-bar/src/ProgressBar.jsx#L47
https://github.com/cerner/terra-core/blob/master/packages/terra-slide-panel/src/SlidePanel.jsx#L74

Last:
https://github.com/cerner/terra-core/blob/master/packages/terra-clinical-item-view/src/Comment.jsx#L41
https://github.com/cerner/terra-core/blob/master/packages/terra-image/src/Image.jsx#L60